inginious/frontend/plugins/task_editorial/pages/task_editorial.py

```diff
@@ -0,0 +1,40 @@
+import os
+
+from inginious.frontend.plugins.multilang.problems.languages import get_all_available_languages
+from inginious.frontend.pages.course_admin.task_edit import CourseEditTask
+
+_TASK_EDITORIAL_TEMPLATE_PATH = os.path.join(os.path.dirname(__file__), "templates")
+
+def check_task_access(task):
+
+    return task.get_accessible_time().is_open()
+
+def editorial_task_tab(course, taskid, task_data, template_helper):
+
+    tab_id = 'tab_editorial'
+    link = '<i class="fa fa-graduation-cap fa-fw"></i>&nbsp; ' + _("Task editorial")
+
+    task_environment = task_data.get('environment')
+
+    if(task_environment == "multiple_languages" or task_environment == "Data Science" or task_environment == "HDL" or task_environment is None):
+        content = template_helper.get_custom_renderer(_TASK_EDITORIAL_TEMPLATE_PATH, layout=False).task_editorial(task_data, get_all_available_languages())
+        return tab_id, link ,content
+    else:
+        return
+
+def editorial_task_preview(course, task, template_helper):
+
+    if(check_task_access(task)):
+        return
+    else:
+        content = template_helper.get_custom_renderer(_TASK_EDITORIAL_TEMPLATE_PATH, layout=False).task_editorial_preview(course, task, get_all_available_languages())
+        return str(content)
+
+def check_editorial_submit(course, taskid, task_data, task_fs):
+
+    task_data['solution_code_language'] = CourseEditTask.dict_from_prefix('solution_code_language',task_data)
+
+    all_languages = get_all_available_languages()
+
+    if not (task_data['solution_code_language'] in all_languages):
+        del task_data['solution_code_language']
```

Comment:
I'm not sure to understand why this is done. Also, do some checks to the task data related to this plugin.